motr/idx.c

```diff
@@ -93,6 +93,65 @@ M0_INTERNAL bool m0__idx_op_invariant(struct m0_op_idx *oi)
 		  m0_op_common_bob_check(&oi->oi_oc));
 }
 
+M0_INTERNAL struct m0_idx*
+m0__idx_entity(struct m0_entity *entity)
+{
+	struct m0_idx *idx;
+
+	M0_PRE(entity != NULL);
+
+	return M0_AMB(idx, entity, in_entity);
+}
+
+M0_INTERNAL int
+m0__idx_pool_version_get(struct m0_idx *idx,
+                         struct m0_pool_version **pv)
+{
+	int               rc;
+	struct m0_client *cinst;
+
+	M0_ENTRY();
+
+	cinst = m0__idx_instance(idx);
+
+	if (pv == NULL)
+		return M0_ERR(-ENOENT);
```

Comment:
I would like to return -EINVAL to say that invalid parameter is passed by caller.

Reply:
And this checking can be done at first before calling "cinst = m0__idx_instance(idx);"
Or maybe we can assert that is not NULL.